### PR TITLE
Splits conversions.rs into separate files

### PR DESCRIPTION
**Pull request by mfaulk**
### Motivation

This is a simple divide-and-conquer play. The conversions.rs file is sort of an "everything and the kitchen sink" file that defines how to convert between "logic" types and gPRC "API" types. The file is large, which makes it hard to navigate and get a sense of what does / doesn't have test coverage.

### In this PR
* Splits conversions.rs into several small files. I think this makes it easier to grok and navigate. There should be no change in actual functionality.

### Future Work
* Evaluate test coverage, use property-based testing to boost test coverage.

### CODE REVIEW

File: api/src/convert/mod.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+//! Conversions between "API types" and "domain/persistence types".
+//!
+//! gRPC and Protobuf provide a reduced selection of types, and so there are some differences between
+//! values stored in the ledger and values transmitted over the API. This module provides conversions
+//! between "equivalent" types, such as `mc_api::blockchain::Block` and `mc_transaction_core::Block`.
+
+mod error;
+
+// blockchain
```
Comment (mfaulk) [inline api/src/convert/mod.rs:11]: Splitting things out as "blockchain" conversions vs. "external" conversions seems to mirror the gRPC definitions.

**Review by rjwalters — APPROVED** [review]
This is nice and makes it really easy to see where unit tests are missing.

Could you please add a ticket for us to follow the same pattern for our other proto APIs as future work?

File: api/src/convert/mod.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+pub use compressed_ristretto::*;
+pub use curve_scalar::*;
+pub use ed25519_signature::*;
+pub use error::ConversionError;
+pub use key_image::*;
+pub use public_address::*;
+pub use ring_mlsag::*;
+pub use ristretto_private::*;
+pub use signature_rct_bulletproofs::*;
+use std::path::PathBuf;
```
Comment (eranrund) [inline api/src/convert/mod.rs:54]: Nit: Separate the `pub` and internal `use` lines

**Review by eranrund — APPROVED** [review]
Nice cleanup!

**Pull request merged**